FantasmoSDK/Classes/FMLocationManager.swift

```diff
@@ -304,16 +305,19 @@ open class FMLocationManager: NSObject, FMApiDelegate {
 extension FMLocationManager : ARSessionDelegate {
     public func session(_ session: ARSession, didUpdate frame: ARFrame) {
         lastFrame = frame
-        arKitTrackingStateStatistics.update(with: frame.camera.trackingState)
         
-        guard state == .localizing else { return }
-        
-        let filterResult = qualityFrameFilter.accepts(frame)
-        frameFailureThrottler.onNext(frameFilterResult: filterResult)
+        if state == .localizing {
+            let filterResult = qualityFrameFilter.accepts(frame)
+            frameFailureThrottler.onNext(frameFilterResult: filterResult)
+            
+            if case .accepted = filterResult {
+                localize(frame: frame, from: session)
+            }
+        }
         
-        if case .accepted = filterResult {
-            localize(frame: frame, from: session)
-        }        
+        if state != .stopped {
```

Comment:
I wonder if the equivalent `state == .localizing || state == .uploading` would be clearer. I could go either way, but just the negated `.stopped` makes me as a reader wonder what the other states are.

Reply:
Negation with `.stopped` has two advantages:
1. It clearly expresses intention that information gathering should be performed only if location manager is not stopped. We don't care about some set of allowed states for gathering, we only care about "stopped" or "not stopped".
2. In case if additional case appear in `FMLocationManager.State` enum there will be problem.

So, in the light of these 2 points I want you to take a final decision. I will implement as you will say.

Reply:
Ok, let's keep `.stopped`.